src/containers/VerletListHelpers.h

```diff
@@ -48,14 +61,17 @@ class VerletListHelpers {
         _verletListsAoS.at(&i).push_back(&j);
     }
 
-    virtual void SoAFunctor(SoA<Particle> &soa, bool /*newton3*/ = true) override {
+    /**
+     * SoAFunctor for verlet list generation. (two cell version)
+     * @param soa the soa
+     */
+    void SoAFunctor(SoA<SoAArraysType> &soa, bool /*newton3*/) override {
       if (soa.getNumParticles() == 0) return;
 
-      Particle **const __restrict__ idptr =
-          reinterpret_cast<Particle **const>(soa.template begin<Particle::AttributeNames::id>());
-      double *const __restrict__ xptr = soa.template begin<Particle::AttributeNames::posX>();
-      double *const __restrict__ yptr = soa.template begin<Particle::AttributeNames::posY>();
-      double *const __restrict__ zptr = soa.template begin<Particle::AttributeNames::posZ>();
+      Particle **const __restrict__ idptr = reinterpret_cast<Particle **const>(soa.begin<AttributeNames::id>());
```

Comment:
`template` in `template begin` no longer needed, as soatype is known for verlet lists and does not depend on particle type